dotcom-rendering/src/lib/article.ts

```diff
@@ -39,6 +40,7 @@ export const enhanceArticleType = (
 		renderingTarget,
 		promotedNewsletter: data.promotedNewsletter,
 		imagesForLightbox,
+		hasAffiliateLinksDisclaimer: !!data.affiliateLinksDisclaimer,
```

Comment:
The truthy value of `hasAffiliateLinksDisclaimer` in the incoming JSON from Frontend determines if we call `enhanceDisclaimer` in `enhanceBlocks`